frontend/js/src/playlists/UserSearch.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import ListenControl from "../listens/ListenControl";
+
+export type UserSearchProps = {
+  userClick: (event: string) => void;
+  placeholder: string;
+};
+
+export type UserSearchState = {
+  newUser: string;
+  userSearchResults: Array<any>;
+};
+
+export default class UserSearch extends React.Component<
+  UserSearchProps,
+  UserSearchState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: UserSearchProps) {
+    super(props);
+    this.state = {
+      newUser: "",
+      userSearchResults: [],
+    };
+  }
+
+  componentDidUpdate(prevProps: UserSearchProps, prevState: UserSearchState) {
+    const { newUser } = this.state;
+
+    if (prevState.newUser !== newUser) {
+      this.searchUsers();
+    }
+  }
```

Comment:
Instead of reacting to a state change this way, we can instead directly call this.searchUsers after setting the state in handleInputChange.
The setState function takes an optional second argument which is a function to be called once the state has been set.
So in the input change handler, you'll be able to simply do this: `this.setState({newUser: event.target.value}, this.searchUsers);` and remove the componentDidUpdate entirely